src/backend/distributed/utils/maintenanced.c

```diff
@@ -292,13 +292,35 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		proc_exit(0);
 	}
 
+	if (myDbData->workerPid != 0 && myDbData->workerPid != MyProcPid)
+	{
+		/*
+		 * Another maintenance daemon is running. This usually happens because
+		 * postgres restarts the daemon after an non-zero exit, and
+		 * InitializeMaintenanceDaemonBackend started one before postgres did.
+		 * In that case, the first one stays and the last one exits.
+		 */
+
+		proc_exit(0);
+	}
+
 	before_shmem_exit(MaintenanceDaemonShmemExit, main_arg);
 
 	Assert(myDbData->workerPid == 0);
```

Comment:
is this assert still valid now with the extra pid check above?
actually it could trigger if `myDbData->workerPid == MyProcPid`.

Guess we need to either remove this assert or add the above equation to it.

Reply:
It is still valid, though the `if` statement is a little misleading because `myDbData->workerPid != 0` is actually sufficient.

Reply:
if you are certain this is ok I am ok with merging.
Feels strange though.

any idea under which circumstances `workerPid == MyPid` is true. It will give an assert in dev. Meaning it should not be possible. However you allude that checking `!= 0` is not enough to check for exit so there is a case where this can happen.

Reply:
I cannot see a situation in which `workerPid == MyPid` because the workerPid always points to a live process and this process did not set workerPid yet. I've removed the check from the `if`